lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -1341,3 +1388,316 @@ def _capture_libvirt_logs(self) -> None:
             )
             with open(str(libvirt_log_local_path), "w") as f:
                 f.write(libvirt_log)
+
+
+class LibvirtDevicePoolImpl(BaseDevicePoolImpl):
+    def __init__(
+        self,
+        node: Union[Node, RemoteNode],
+        runbook: BaseLibvirtPlatformSchema,
+    ) -> None:
+        # Mapping of Host Device Passthrough
+        self.AVAILABLE_HOST_DEVICES: Dict[
+            HostDevicePoolType, Dict[str, List[DeviceAddressSchema]]
+        ] = {}
+        self.SUPPORTED_HOST_DEVICE_POOLTYPE = [
+            HostDevicePoolType.PCI_NIC,
+            HostDevicePoolType.PCI_GPU,
+        ]
+        self.POOL_TYPE_TO_DEVICE_PROPERTY = {
+            HostDevicePoolType.PCI_NIC: DeviceAddressSchema,
+            HostDevicePoolType.PCI_GPU: DeviceAddressSchema,
+        }
+        self.host_node = node
+        self.platform_runbook = runbook
+
+    def _configure_device_passthrough_pool(
+        self,
+        device_configs: Optional[List[HostDevicePoolSchema]],
+        supported_pool_type: Optional[List[HostDevicePoolType]],
+    ) -> None:
+        if device_configs:
+            # Check if host support device passthrough
+            self._check_passthrough_support(self.host_node)
+
+            if not supported_pool_type:
+                supported_pool_type = self.SUPPORTED_HOST_DEVICE_POOLTYPE
```

Comment:
Should this be a set intersection between `supported_pool_type` and `SUPPORTED_HOST_DEVICE_POOLTYPE`?

Reply:
@cwize1

Idea is to implement device pool type support at platform specific class. If not provided, we will take what we have as statically defined